File: @LV1/1.3/fizzbuzz/fizzbuzz.rb
def fizzbuzz(n, x, y)
  mx = n % x == 0
  my = n % y == 0

  case
    when (mx and my) then "FizzBuzz"
    when mx then "Fizz"
    when my then "Buzz"
    else n
  end
end

x, y, n = gets.split.map(&:to_i)

(1..n).each { |i| puts fizzbuzz(i, x, y) }

File: @LV1/1.3/fizzbuzz/fizzbuzz.rs
use std::io;

fn main() {
    let stdin = io::stdin();
    let mut buff = String::new();

    stdin.read_line(&mut buff).unwrap();
    let num: Vec<i32> = buff
        .split_whitespace()
        .map(|n| n.parse().unwrap())
        .collect();

    let x = num[0];
    let y = num[1];
    let n = num[2];

    for i in 1..(n + 1) {
        match (i % x == 0, i % y == 0) {
            (true, true) => println!("FizzBuzz"),
            (true, _) => println!("Fizz"),
            (_, true) => println!("Buzz"),
            (_, _) => println!("{}", i),
        }
    }
}
